cpp/include/Ice/LocalException.h

```diff
@@ -1,30 +1,54 @@
-//
-// Copyright (c) ZeroC, Inc. All rights reserved.
-//
+// Copyright (c) ZeroC, Inc.
 
 #ifndef ICE_LOCAL_EXCEPTION_H
 #define ICE_LOCAL_EXCEPTION_H
 
 #include "Exception.h"
 
+#include <memory>
+#include <string>
+#include <vector>
+
 namespace Ice
 {
-    /**
-     * Base class for all Ice exceptions not defined in Slice.
-     * \headerfile Ice/Ice.h
-     */
+    /// Base class for all Ice exceptions not defined in Slice.
+    /// \headerfile Ice/Ice.h
     class ICE_API LocalException : public Exception
     {
     public:
-        /**
-         * Constructs a local exception.
-         * @param file The file where this exception is constructed. This C string is not copied.
-         * @param line The line where this exception is constructed.
-         * @param message The error message adopted by this exception and returned by what().
-         */
-        LocalException(const char* file, int line, std::string message) : Exception(file, line, std::move(message)) {}
+        /// Constructs a local exception.
+        /// @param file The file where this exception is constructed. This C string is not copied.
+        /// @param line The line where this exception is constructed.
+        /// @param message The error message adopted by this exception and returned by what().
+        LocalException(const char* file, int line, std::string message);
+
+        [[nodiscard]] const char* what() const noexcept final;
```

Comment:
I know that `what` is pretty standard C++ exception stuff, but should we carry over the doc-comment for it?
```
        /**
         * Returns the error message of this exception.
         * @return The error message.
         */
        [[nodiscard]] const char* what() const noexcept override;
```